ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java

```diff
@@ -0,0 +1,190 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import static java.util.stream.Collectors.toUnmodifiableList;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.BlockParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.ImmutableFeeHistoryResult;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class EthFeeHistory implements JsonRpcMethod {
+  private final ProtocolSchedule protocolSchedule;
+  private final BlockchainQueries blockchainQueries;
+  private final Blockchain blockchain;
+
+  public EthFeeHistory(
+      final ProtocolSchedule protocolSchedule, final BlockchainQueries blockchainQueries) {
+    this.protocolSchedule = protocolSchedule;
+    this.blockchainQueries = blockchainQueries;
+    this.blockchain = blockchainQueries.getBlockchain();
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_FEE_HISTORY.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext request) {
+    final Object requestId = request.getRequest().getId();
+
+    final long blockCount = request.getRequiredParameter(0, Long.class);
+    if (blockCount < 1 || blockCount > 1024) {
+      return new JsonRpcErrorResponse(requestId, JsonRpcError.INVALID_PARAMS);
+    }
+    final BlockParameter highestBlock = request.getRequiredParameter(1, BlockParameter.class);
+    final Optional<List<Double>> maybeRewardPercentiles =
+        request.getOptionalParameter(2, Double[].class).map(Arrays::asList);
+
+    final long chainHeadBlockNumber = blockchain.getChainHeadBlockNumber();
+    final long resolvedHighestBlockNumber =
+        highestBlock
+            .getNumber()
+            .orElse(
+                highestBlock.isEarliest()
```

Comment:
why this check ? if it's earlist the highestBlock.getNumber() will not return automaticaly the genesis ? https://github.com/hyperledger/besu/blob/e230a445bb457bbac27bf4c384da449ccbd3f043/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/parameters/BlockParameter.java#L42

Reply:
Ah yes, so I just need to check for the latest and pending cases.